app/models/category.rb

```diff
@@ -1,3 +1,15 @@
 class Category < ApplicationRecord
   has_many :product, dependent: :destroy
+
+  scope :sort_by_name, ->{order(name: :asc)}
+  scope :search_by_name, lambda {|query|
+    where("name LIKE ?", "%#{query}%")
```

Comment:
```suggestion
    where("name LIKE ?", "%#{query}%") if query
```

Như này đc ko em. Bên trên controller gọi chained scope luôn cx đc